File: python/packages/wetwire-agent/tests/domains/aws/scenarios/s3_log_bucket/expected/storage.py
"""Storage resources: LogBucket - S3 bucket for application logs."""

from . import *  # noqa: F403


class LogBucketServerSideEncryptionByDefault:
    """Server-side encryption configuration for log bucket."""

    resource: s3.bucket.ServerSideEncryptionByDefault
    sse_algorithm = s3.ServerSideEncryption.AES256


class LogBucketServerSideEncryptionRule:
    """Encryption rule for log bucket."""

    resource: s3.bucket.ServerSideEncryptionRule
    server_side_encryption_by_default = LogBucketServerSideEncryptionByDefault


class LogBucketEncryption:
    """Bucket encryption configuration."""

    resource: s3.bucket.BucketEncryption
    server_side_encryption_configuration = [LogBucketServerSideEncryptionRule]


class LogBucketPublicAccessBlock:
    """Block all public access to the log bucket."""

    resource: s3.bucket.PublicAccessBlockConfiguration
    block_public_acls = True
    block_public_policy = True
    ignore_public_acls = True
    restrict_public_buckets = True


class LogBucket:
    """S3 bucket for storing application logs."""

    resource: s3.Bucket
    bucket_encryption = LogBucketEncryption
    public_access_block_configuration = LogBucketPublicAccessBlock

File: go/wetwire-agent/testdata/scenarios/s3_log_bucket/expected/storage.go
// Package s3_log_bucket contains CloudFormation resources for log storage.
// Storage resources: LogBucket - S3 bucket for application logs.
package s3_log_bucket

import (
	"github.com/lex00/wetwire/go/wetwire-aws/resources/s3"
)

// LogBucketServerSideEncryptionByDefault configures AES-256 encryption.
var LogBucketServerSideEncryptionByDefault = &s3.Bucket_ServerSideEncryptionByDefault{
	SSEAlgorithm: "AES256",
}

// LogBucketServerSideEncryptionRule wraps the encryption configuration.
var LogBucketServerSideEncryptionRule = s3.Bucket_ServerSideEncryptionRule{
	ServerSideEncryptionByDefault: LogBucketServerSideEncryptionByDefault,
}

// LogBucketEncryption configures bucket encryption with AES-256.
var LogBucketEncryption = &s3.Bucket_BucketEncryption{
	ServerSideEncryptionConfiguration: []s3.Bucket_ServerSideEncryptionRule{
		LogBucketServerSideEncryptionRule,
	},
}

// LogBucketPublicAccessBlock blocks all public access to the log bucket.
var LogBucketPublicAccessBlock = &s3.Bucket_PublicAccessBlockConfiguration{
	BlockPublicAcls:       true,
	BlockPublicPolicy:     true,
	IgnorePublicAcls:      true,
	RestrictPublicBuckets: true,
}

// LogBucket is an S3 bucket for storing application logs.
var LogBucket = s3.Bucket{
	BucketEncryption:               LogBucketEncryption,
	PublicAccessBlockConfiguration: LogBucketPublicAccessBlock,
}
